### PR TITLE
Split OTLP endpoint path to fix Langfuse/LangSmith integration

### LINKED ISSUE #4811: OTLP endpoint with custom path gets URL-encoded, breaking Langfuse/LangSmith integration
## Bug description

The OTLP HTTP exporters use `otlptracehttp.WithEndpoint()` and `otlpmetrichttp.WithEndpoint()`, which expect a **host:port only**. When users include a path in the endpoint (required by platforms like Langfuse and LangSmith), the path component gets URL-encoded and the SDK appends its default `/v1/traces` suffix to the mangled URL.

This affects both MCPTelemetryConfig (operator) and inline telemetry config (CLI).

## Steps to reproduce

1. Create an MCPTelemetryConfig with a Langfuse endpoint:
   ```yaml
   apiVersion: toolhive.stacklok.com/v1alpha1
   kind: MCPTelemetryConfig
   metadata:
     name: langfuse
   spec:
     openTelemetry:
       enabled: true
       endpoint: https://cloud.langfuse.com/api/public/otel
       tracing:
         enabled: true
       headers:
         Authorization: "Basic <base64>"
   ```
2. The operator's `NormalizeTelemetryConfig` strips the scheme, producing `cloud.langfuse.com/api/public/otel`
3. This is passed to `otlptracehttp.WithEndpoint("cloud.langfuse.com/api/public/otel")`
4. The Go OTLP SDK treats the entire string as the hostname, URL-encodes the slashes, and appends `/v1/traces`

## Expected behavior

The exporter should send traces to:
```
https://cloud.langfuse.com/api/public/otel/v1/traces
```

## Actual behavior

The exporter sends traces to:
```
https://cloud.langfuse.com%2Fapi%2Fpublic%2Fotel/v1/traces
```

This fails for any OTLP backend that uses a non-default base path, including:
- **Langfuse**: expects `/api/public/otel/v1/traces`
- **LangSmith**: expects `/api/v1/otel/v1/traces`
- Any self-hosted collector behind a reverse proxy with a path prefix

## Root cause

In `pkg/telemetry/providers/otlp/tracing.go` and `metrics.go`, the endpoint is passed directly to `WithEndpoint()` without separating the host from any path component:

```go
// tracing.go:19
otlptracehttp.WithEndpoint(config.Endpoint),

// metrics.go:31
otlpmetrichttp.WithEndpoint(config.Endpoint),
```

The Go OTLP HTTP SDK's `WithEndpoint` only accepts `host:port`. To customize the URL path, `WithURLPath()` must be used separately.

## Proposed fix

Parse the endpoint to separate host:port from any path, then use both options:

```go
host, basePath := splitEndpointPath(config.Endpoint)
opts := []otlptracehttp.Option{
    otlptracehttp.WithEndpoint(host),
}
if basePath != "" {
    opts = append(opts, otlptracehttp.WithURLPath(basePath+"/v1/traces"))
}
```

And the equivalent for metrics with `/v1/metrics`.

### Files to change

- `pkg/telemetry/providers/otlp/tracing.go` — `createTraceExporter`: split endpoint, add `WithURLPath`
- `pkg/telemetry/providers/otlp/metrics.go` — `createMetricExporter`: same treatment
- Add a `splitEndpointPath` helper (e.g., in `pkg/telemetry/providers/otlp/config.go`)

No CRD schema changes are needed — the existing `endpoint` field is sufficient once paths are handled correctly.

## Environment

- ToolHive: current main branch
- Affects: MCPTelemetryConfig CRD and inline `spec.telemetry` on MCPServer/VirtualMCPServer
- Go OTLP SDK: `go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracehttp`
- Reported by a user integrating with Langfuse (`cloud.langfuse.com/api/public/otel`) and LangSmith (`smith.langchain.com/api/v1/otel`)

## Additional context

The user attempted to work around this by including the full path in the endpoint field, but the SDK URL-encodes it. There is no current workaround short of placing an OTEL Collector in front of the backend to rewrite the path.

### PR DESCRIPTION
**Pull request by ChrisJBurns**
## Summary

- The Go OTLP HTTP SDK's `WithEndpoint()` only accepts `host:port`, but platforms like Langfuse and LangSmith require a custom base path in the endpoint URL. When users configure an endpoint like `https://cloud.langfuse.com/api/public/otel`, the path component gets URL-encoded (slashes become `%2F`), breaking all requests to these backends.
- Parse the endpoint string into `host:port` and path components using a new `splitEndpointPath` helper, then pass the host to `WithEndpoint()` and the path (with signal suffix appended) to `WithURLPath()`.

Fixes #4811

## Type of change

- [x] Bug fix

## Test plan

- [x] Unit tests (`task test`)
- [x] Linting (`task lint-fix`)
- [x] Manual testing (describe below)

### Manual verification with Langfuse on Kind

Deployed a full Langfuse v3 stack (PostgreSQL, ClickHouse, Redis, MinIO) into a Kind cluster alongside the ToolHive operator built from this branch. Created an `MCPTelemetryConfig` with the exact scenario from the bug report — an endpoint with a custom path:

```yaml
endpoint: "http://langfuse.langfuse.svc.cluster.local:3000/api/public/otel"
```

**Before the fix**, the proxy runner would log:
```
traces export: failed to send to http://langfuse...%2Fapi%2Fpublic%2Fotel/v1/traces
```

**After the fix**, traces export silently (no error), and OTLP trace JSON files appear in MinIO:
```
[2026-04-14 16:50:57 UTC] 1.9KiB otel/test-project/.../0ef0144d-...json
[2026-04-14 16:59:24 UTC] 1.9KiB otel/test-project/.../349cd6c0-...json
```

The trace payload contains correct span data with `service.name: "echo-server"`, MCP-specific attributes (`mcp.server.name`, `mcp.transport`), and standard OTEL attributes.

<details>
<summary>Full reproduction steps</summary>

#### Prerequisites

- `kind`, `kubectl`, `task`, `helm` installed
- Clone this branch

#### 1. Create cluster and deploy operator

```bash
task kind-setup-e2e
task operator-install-crds
task operator-deploy-local
```

#### 2. Create namespaces

```bash
KUBECONFIG=kconfig.yaml kubectl create namespace langfuse
KUBECONFIG=kconfig.yaml kubectl create namespace mcp-test
```

#### 3. Deploy Langfuse stack

Apply each manifest and wait for readiness:

```bash
KUBECONFIG=kconfig.yaml kubectl apply -f test/manual/langfuse/postgres.yaml
KUBECONFIG=kconfig.yaml kubectl wait --for=condition=ready pod -l app=postgres -n langfuse --timeout=120s

KUBECONFIG=kconfig.yaml kubectl apply -f test/manual/langfuse/redis.yaml
KUBECONFIG=kconfig.yaml kubectl wait --for=condition=ready pod -l app=redis -n langfuse --timeout=60s

KUBECONFIG=kconfig.yaml kubectl apply -f test/manual/langfuse/clickhouse.yaml
KUBECONFIG=kconfig.yaml kubectl wait --for=condition=ready pod -l app=clickhouse -n langfuse --timeout=120s

KUBECONFIG=kconfig.yaml kubectl apply -f test/manual/langfuse/minio.yaml
KUBECONFIG=kconfig.yaml kubectl wait --for=condition=ready pod -l app=minio -n langfuse --timeout=120s

KUBECONFIG=kconfig.yaml kubectl apply -f test/manual/langfuse/langfuse.yaml
KUBECONFIG=kconfig.yaml kubectl rollout status deployment/langfuse -n langfuse --timeout=300s
```

#### 4. Deploy MCPServer with Langfuse telemetry

```bash
KUBECONFIG=kconfig.yaml kubectl apply -f test/manual/langfuse/mcp-with-langfuse.yaml
sleep 30
KUBECONFIG=kconfig.yaml kubectl get pods -n mcp-test
```

#### 5. Send traffic and verify

```bash
# Send requests through the MCP proxy
for i in $(seq 1 5); do
  KUBECONFIG=kconfig.yaml kubectl run -n mcp-test "curl-$i" \
    --image=curlimages/curl --restart=Never --rm -it -- \
    curl -s -o /dev/null -w "%{http_code}\n" http://mcp-echo-server-proxy:8080/health
done

# Wait for OTLP batch flush and check proxy logs (no export errors = success)
sleep 15
PROXY=$(KUBECONFIG=kconfig.yaml kubectl get pods -n mcp-test -l app=mcpserver -o jsonpath='{.items[0].metadata.name}')
KUBECONFIG=kconfig.yaml kubectl logs -n mcp-test "$PROXY" -c toolhive --tail=5

# Verify traces landed in MinIO
KUBECONFIG=kconfig.yaml kubectl create -n langfuse -f - <<'EOF'
apiVersion: v1
kind: Pod
metadata:
  name: mc-check
spec:
  restartPolicy: Never
  containers:
  - name: mc
    image: minio/mc
    command: ["sh", "-c", "mc alias set m http://minio:9000 minioadmin minioadmin && mc ls --recursive m/langfuse/ && echo DONE"]
EOF

sleep 15
KUBECONFIG=kconfig.yaml kubectl logs -n langfuse mc-check
KUBECONFIG=kconfig.yaml kubectl delete pod mc-check -n langfuse
```

#### 6. Teardown

```bash
task kind-destroy
```

</details>

<details>
<summary>Manifest: postgres.yaml</summary>

```yaml
apiVersion: v1
kind: Service
metadata:
  name: postgres
  namespace: langfuse
spec:
  selector:
    app: postgres
  ports:
    - port: 5432
      targetPort: 5432
---
apiVersion: apps/v1
kind: Deployment
metadata:
  name: postgres
  namespace: langfuse
spec:
  replicas: 1
  selector:
    matchLabels:
      app: postgres
  template:
    metadata:
      labels:
        app: postgres
    spec:
      containers:
        - name: postgres
          image: postgres:16-alpine
          ports:
            - containerPort: 5432
          env:
            - name: POSTGRES_DB
              value: langfuse
            - name: POSTGRES_USER
              value: langfuse
            - name: POSTGRES_PASSWORD
              value: langfuse
          readinessProbe:
            exec:
              command: ["pg_isready", "-U", "langfuse"]
            initialDelaySeconds: 5
            periodSeconds: 5
```

</details>

<details>
<summary>Manifest: redis.yaml</summary>

```yaml
apiVersion: v1
kind: Service
metadata:
  name: redis
  namespace: langfuse
spec:
  selector:
    app: redis
  ports:
    - port: 6379
      targetPort: 6379
---
apiVersion: apps/v1
kind: Deployment
metadata:
  name: redis
  namespace: langfuse
spec:
  replicas: 1
  selector:
    matchLabels:
      app: redis
  template:
    metadata:
      labels:
        app: redis
    spec:
      containers:
        - name: redis
          image: redis:7-alpine
          args: ["--maxmemory-policy", "noeviction"]
          ports:
            - containerPort: 6379
          readinessProbe:
            exec:
              command: ["redis-cli", "ping"]
            initialDelaySeconds: 5
            periodSeconds: 5
```

</details>

<details>
<summary>Manifest: clickhouse.yaml</summary>

```yaml
apiVersion: v1
kind: Service
metadata:
  name: clickhouse
  namespace: langfuse
spec:
  selector:
    app: clickhouse
  ports:
    - name: http
      port: 8123
      targetPort: 8123
    - name: native
      port: 9000
      targetPort: 9000
---
apiVersion: apps/v1
kind: Deployment
metadata:
  name: clickhouse
  namespace: langfuse
spec:
  replicas: 1
  selector:
    matchLabels:
      app: clickhouse
  template:
    metadata:
      labels:
        app: clickhouse
    spec:
      containers:
        - name: clickhouse
          image: clickhouse/clickhouse-server:24-alpine
          ports:
            - containerPort: 8123
            - containerPort: 9000
          env:
            - name: CLICKHOUSE_DB
              value: default
            - name: CLICKHOUSE_USER
              value: default
            - name: CLICKHOUSE_DEFAULT_ACCESS_MANAGEMENT
              value: "1"
            - name: CLICKHOUSE_PASSWORD
              value: clickhouse
          readinessProbe:
            httpGet:
              path: /ping
              port: 8123
            initialDelaySeconds: 10
            periodSeconds: 5
```

</details>

<details>
<summary>Manifest: minio.yaml</summary>

```yaml
apiVersion: v1
kind: Service
metadata:
  name: minio
  namespace: langfuse
spec:
  selector:
    app: minio
  ports:
    - name: api
      port: 9000
      targetPort: 9000
    - name: console
      port: 9001
      targetPort: 9001
---
apiVersion: apps/v1
kind: Deployment
metadata:
  name: minio
  namespace: langfuse
spec:
  replicas: 1
  selector:
    matchLabels:
      app: minio
  template:
    metadata:
      labels:
        app: minio
    spec:
      containers:
        - name: minio
          image: minio/minio:latest
          args: ["server", "/data", "--console-address", ":9001"]
          ports:
            - containerPort: 9000
            - containerPort: 9001
          env:
            - name: MINIO_ROOT_USER
              value: "minioadmin"
            - name: MINIO_ROOT_PASSWORD
              value: "minioadmin"
          readinessProbe:
            httpGet:
              path: /minio/health/ready
              port: 9000
            initialDelaySeconds: 5
            periodSeconds: 5
---
apiVersion: batch/v1
kind: Job
metadata:
  name: minio-create-bucket
  namespace: langfuse
spec:
  template:
    spec:
      restartPolicy: OnFailure
      containers:
        - name: mc
          image: minio/mc:latest
          command:
            - /bin/sh
            - -c
            - |
              sleep 5
              mc alias set myminio http://minio:9000 minioadmin minioadmin
              mc mb --ignore-existing myminio/langfuse
              echo "Bucket created"
```

</details>

<details>
<summary>Manifest: langfuse.yaml</summary>

```yaml
apiVersion: v1
kind: Service
metadata:
  name: langfuse
  namespace: langfuse
spec:
  selector:
    app: langfuse
  ports:
    - name: http
      port: 3000
      targetPort: 3000
---
apiVersion: apps/v1
kind: Deployment
metadata:
  name: langfuse
  namespace: langfuse
spec:
  replicas: 1
  selector:
    matchLabels:
      app: langfuse
  template:
    metadata:
      labels:
        app: langfuse
    spec:
      containers:
        - name: langfuse
          image: langfuse/langfuse:3
          ports:
            - containerPort: 3000
          env:
            - name: DATABASE_URL
              value: "postgresql://langfuse:langfuse@postgres:5432/langfuse"
            - name: DIRECT_URL
              value: "postgresql://langfuse:langfuse@postgres:5432/langfuse"
            - name: CLICKHOUSE_CLUSTER_ENABLED
              value: "false"
            - name: CLICKHOUSE_URL
              value: "http://clickhouse:8123"
            - name: CLICKHOUSE_MIGRATION_URL
              value: "clickhouse://clickhouse:9000"
            - name: CLICKHOUSE_DB
              value: "default"
            - name: CLICKHOUSE_USER
              value: "default"
            - name: CLICKHOUSE_PASSWORD
              value: "clickhouse"
            - name: REDIS_HOST
              value: "redis"
            - name: REDIS_PORT
              value: "6379"
            - name: REDIS_AUTH
              value: ""
            - name: REDIS_CONNECTION_STRING
              value: "redis://redis:6379/0"
            - name: LANGFUSE_S3_EVENT_UPLOAD_BUCKET
              value: "langfuse"
            - name: LANGFUSE_S3_EVENT_UPLOAD_REGION
              value: "us-east-1"
            - name: LANGFUSE_S3_EVENT_UPLOAD_ACCESS_KEY_ID
              value: "minioadmin"
            - name: LANGFUSE_S3_EVENT_UPLOAD_SECRET_ACCESS_KEY
              value: "minioadmin"
            - name: LANGFUSE_S3_EVENT_UPLOAD_ENDPOINT
              value: "http://minio:9000"
            - name: LANGFUSE_S3_EVENT_UPLOAD_FORCE_PATH_STYLE
              value: "true"
            - name: LANGFUSE_S3_MEDIA_UPLOAD_BUCKET
              value: "langfuse"
            - name: LANGFUSE_S3_MEDIA_UPLOAD_REGION
              value: "us-east-1"
            - name: LANGFUSE_S3_MEDIA_UPLOAD_ACCESS_KEY_ID
              value: "minioadmin"
            - name: LANGFUSE_S3_MEDIA_UPLOAD_SECRET_ACCESS_KEY
              value: "minioadmin"
            - name: LANGFUSE_S3_MEDIA_UPLOAD_ENDPOINT
              value: "http://minio:9000"
            - name: LANGFUSE_S3_MEDIA_UPLOAD_FORCE_PATH_STYLE
              value: "true"
            - name: NEXTAUTH_URL
              value: "http://langfuse.langfuse.svc.cluster.local:3000"
            - name: NEXTAUTH_SECRET
              value: "test-secret-for-kind-cluster-only"
            - name: SALT
              value: "test-salt-for-kind-cluster-only"
            - name: ENCRYPTION_KEY
              value: "0000000000000000000000000000000000000000000000000000000000000000"
            - name: LANGFUSE_INIT_ORG_ID
              value: "test-org"
            - name: LANGFUSE_INIT_ORG_NAME
              value: "Test Org"
            - name: LANGFUSE_INIT_PROJECT_ID
              value: "test-project"
            - name: LANGFUSE_INIT_PROJECT_NAME
              value: "Test Project"
            - name: LANGFUSE_INIT_PROJECT_PUBLIC_KEY
              value: "pk-lf-test-public-key"
            - name: LANGFUSE_INIT_PROJECT_SECRET_KEY
              value: "sk-lf-test-secret-key"
            - name: LANGFUSE_INIT_USER_EMAIL
              value: "admin@test.local"
            - name: LANGFUSE_INIT_USER_NAME
              value: "Admin"
            - name: LANGFUSE_INIT_USER_PASSWORD
              value: "password"
          readinessProbe:
            httpGet:
              path: /api/public/health
              port: 3000
            initialDelaySeconds: 60
            periodSeconds: 10
            timeoutSeconds: 5
            failureThreshold: 12
          resources:
            requests:
              memory: "512Mi"
              cpu: "250m"
            limits:
              memory: "1Gi"
```

</details>

<details>
<summary>Manifest: mcp-with-langfuse.yaml</summary>

```yaml
---
apiVersion: toolhive.stacklok.dev/v1alpha1
kind: MCPTelemetryConfig
metadata:
  name: langfuse-otel
  namespace: mcp-test
spec:
  openTelemetry:
    enabled: true
    # This is the key: endpoint WITH a path — the bug in #4811
    endpoint: "http://langfuse.langfuse.svc.cluster.local:3000/api/public/otel"
    insecure: true
    tracing:
      enabled: true
      samplingRate: "1.0"
    headers:
      Authorization: "Basic cGstbGYtdGVzdC1wdWJsaWMta2V5OnNrLWxmLXRlc3Qtc2VjcmV0LWtleQ=="
---
apiVersion: toolhive.stacklok.dev/v1alpha1
kind: MCPServer
metadata:
  name: echo-server
  namespace: mcp-test
spec:
  image: "ghcr.io/stackloklabs/yardstick/yardstick-server:1.1.1"
  transport: sse
  telemetryConfigRef:
    name: langfuse-otel
```

</details>

## Changes

| File | Change |
|------|--------|
| `pkg/telemetry/providers/otlp/endpoint.go` | New `splitEndpointPath` helper that separates `host:port` from path |
| `pkg/telemetry/providers/otlp/endpoint_test.go` | Table-driven tests for the helper (Langfuse, LangSmith, trailing slash, etc.) |
| `pkg/telemetry/providers/otlp/tracing.go` | Use `splitEndpointPath`, add `WithURLPath(basePath+"/v1/traces")` |
| `pkg/telemetry/providers/otlp/metrics.go` | Use `splitEndpointPath`, add `WithURLPath(basePath+"/v1/metrics")` |
| `pkg/telemetry/providers/otlp/tracing_test.go` | Add test case for endpoint with custom path |
| `pkg/telemetry/providers/otlp/metrics_test.go` | Add test cases for endpoint with custom path |

## Does this introduce a user-facing change?

Yes. Users can now configure OTLP endpoints with custom base paths (e.g., `https://cloud.langfuse.com/api/public/otel`) in both `MCPTelemetryConfig` CRDs and inline telemetry config. Previously, any path in the endpoint was URL-encoded, breaking integration with Langfuse, LangSmith, and collectors behind reverse proxies with path prefixes.

## Implementation plan

<details>
<summary>Approved implementation plan</summary>

1. Add `splitEndpointPath` helper in `pkg/telemetry/providers/otlp/endpoint.go` that splits a scheme-stripped endpoint into `host:port` and base path
2. In `tracing.go`, call `splitEndpointPath` and conditionally add `WithURLPath(basePath+"/v1/traces")`
3. In `metrics.go`, same treatment with `WithURLPath(basePath+"/v1/metrics")`
4. Add comprehensive tests for the helper and new exporter behavior

Key SDK details: `WithURLPath()` replaces the entire URL path (does not append to default), so we must manually construct `basePath + "/v1/traces"` and `basePath + "/v1/metrics"`. The SDK's `cleanPath()` normalizes the path afterward.

</details>

Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: pkg/telemetry/providers/otlp/endpoint.go [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+// SPDX-License-Identifier: Apache-2.0
+
+package otlp
+
+import "strings"
+
+// splitEndpointPath separates an OTLP endpoint string into its host:port and
+// path components. The endpoint is expected to have its scheme already stripped
+// (by NormalizeTelemetryConfig). If no path is present, basePath is empty.
+func splitEndpointPath(endpoint string) (hostPort, basePath string) {
```
Comment (ChrisJBurns) [inline pkg/telemetry/providers/otlp/endpoint.go:28]: **[MEDIUM] splitEndpointPath not defensive against scheme-prefixed endpoints** (Consensus: 7/10)

The doc comment correctly states the scheme should be stripped upstream, but `NormalizeTelemetryConfig` only runs in the operator path. The CLI `--otel-endpoint` flag passes endpoints through without scheme stripping. If a CLI user passes `https://cloud.langfuse.com/api/public/otel`, `strings.Index` finds the first `/` inside `://`, producing `hostPort="https:"` — completely wrong.

This is a **pre-existing issue** (the old code also broke on scheme-prefixed CLI endpoints) and shouldn't block this PR. Consider a follow-up to either add defensive scheme stripping here or normalize in the CLI path.

*Raised by: go-expert-developer, code-reviewer, toolhive-expert*

File: pkg/telemetry/providers/otlp/endpoint_test.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+			endpoint:     "cloud.langfuse.com/api/public/otel/",
+			wantHostPort: "cloud.langfuse.com",
+			wantBasePath: "/api/public/otel",
+		},
+		{
+			name:         "empty string",
+			endpoint:     "",
+			wantHostPort: "",
+			wantBasePath: "",
+		},
```
Comment (ChrisJBurns) [inline pkg/telemetry/providers/otlp/endpoint_test.go:80]: **[LOW] Missing test case for bare trailing slash** (Consensus: 9/10)

Consider adding a test case for `"localhost:4318/"` (host:port with bare trailing slash, no path). `TrimRight("/", "/")` correctly returns `""`, so `WithURLPath` is not called — this edge case is handled correctly but should be documented in tests.

```suggestion
		},
		{
			name:         "host:port with trailing slash only",
			endpoint:     "localhost:4318/",
			wantHostPort: "localhost:4318",
			wantBasePath: "",
		},
```

*Raised by: site-reliability-engineer, go-expert-developer, toolhive-expert*

File: pkg/telemetry/providers/otlp/tracing.go [hunk 3]
```diff
@@ -15,8 +15,13 @@ import (
...
 
 func createTraceExporter(ctx context.Context, config Config) (sdktrace.SpanExporter, error) {
+	host, basePath := splitEndpointPath(config.Endpoint)
 	opts := []otlptracehttp.Option{
-		otlptracehttp.WithEndpoint(config.Endpoint),
+		otlptracehttp.WithEndpoint(host),
+	}
+
+	if basePath != "" {
+		opts = append(opts, otlptracehttp.WithURLPath(basePath+"/v1/traces"))
```
Comment (ChrisJBurns) [inline pkg/telemetry/providers/otlp/tracing.go]: **[LOW] Hardcoded OTLP signal path suffix** (Consensus: 10/10)

The `"/v1/traces"` and `"/v1/metrics"` paths (in `metrics.go`) are the OTLP HTTP spec defaults, hardcoded inline. Consider extracting as package-level constants for maintainability — e.g., in `endpoint.go`:

```go
const (
    otlpTracesPath  = "/v1/traces"
    otlpMetricsPath = "/v1/metrics"
)
```

Minor nit, not blocking.

*Raised by: all 4 agents*

File: pkg/telemetry/providers/otlp/metrics_test.go [hunk 4]
```diff
@@ -37,13 +37,13 @@ func TestCreateMetricExporter(t *testing.T) {
 			wantErr: false,
 		},
 		{
-			name: "error creating metrics exporter due to malformed endpoint",
+			name: "endpoint with custom path",
```
Comment (ChrisJBurns) [inline pkg/telemetry/providers/otlp/metrics_test.go:43]: **[MEDIUM] Removed error tests leave zero negative coverage** (Consensus: 10/10)

The new "endpoint with custom path" test cases are valuable additions. However, they replaced the only `wantErr: true` cases in both `TestCreateMetricExporter` and `TestNewMetricReader`. The tracing tests retain a cancelled-context error case, but metrics now has none.

Consider adding back at least one error-path test — for example, a cancelled-context case like the tracing tests have:

```go
{
    name: "error creating metrics exporter due to cancelled context",
    config: Config{
        Endpoint: "localhost:4318",
        Insecure: true,
    },
    // Use a cancelled context to trigger an error
    wantErr: true,
    errMsg:  "context canceled",
},
```

(Note: `createMetricExporter` doesn't take a context-factory like the tracing tests, so the setup would need a small adjustment.)

*Raised by: all 4 agents*

**Review by ChrisJBurns — COMMENTED** [review]
## Multi-Agent Consensus Review
**Agents consulted**: site-reliability-engineer, go-expert-developer, code-reviewer, toolhive-expert

### Consensus Summary
| \# | Finding | Consensus | Severity | Action |
|---|---------|-----------|----------|--------|
| 1 | splitEndpointPath not defensive against scheme-prefixed endpoints | 7/10 | MEDIUM | Discuss |
| 2 | Missing test case for bare trailing slash endpoint | 9/10 | LOW | Fix |
| 3 | Hardcoded signal path suffixes | 10/10 | LOW | Fix |
| 4 | Removed error tests reduce negative coverage | 10/10 | MEDIUM | Fix |

### Overall
This is a well-crafted, focused bug fix with thorough manual verification against Langfuse on Kind. The approach is sound: `splitEndpointPath` correctly handles the OTLP HTTP SDK's expectation of `host:port` in `WithEndpoint()` while preserving custom URL paths via `WithURLPath()`. Backward compatibility is maintained — existing setups without custom paths are completely unaffected because `WithURLPath` is only called when `basePath != ""`.

All four agents unanimously agree the core approach is correct. The findings are primarily about test hardening: restoring error-path coverage that was removed (F4, unanimous), adding an edge-case test for bare trailing slashes (F2), and a minor code hygiene suggestion to extract hardcoded OTLP path constants (F3). The one MEDIUM finding about scheme handling (F1) is a pre-existing issue in the CLI path that predates this PR — it's worth a follow-up but should not block merging.

---
*Generated with Claude Code*

File: pkg/telemetry/providers/otlp/endpoint.go (hunk 1 above)
Comment (ChrisJBurns) [reply pkg/telemetry/providers/otlp/endpoint.go:28]: Addressed in f5e03cd3. `splitEndpointPath` now defensively strips `http://` and `https://` prefixes before splitting, so the CLI path (`thv run --otel-endpoint`) works correctly even without upstream normalization. Added test cases for `https://cloud.langfuse.com/api/public/otel`, `http://localhost:4318`, and `https://api.honeycomb.io` to cover this.

File: pkg/telemetry/providers/otlp/endpoint_test.go (hunk 2 above)
Comment (ChrisJBurns) [reply pkg/telemetry/providers/otlp/endpoint_test.go:80]: Added in f5e03cd3. Test case for `"localhost:4318/"` confirms `basePath` is empty (trailing-slash-only is not a path).

File: pkg/telemetry/providers/otlp/tracing.go (hunk 3 above)
Comment (ChrisJBurns) [reply pkg/telemetry/providers/otlp/tracing.go]: Done in f5e03cd3. Extracted as `otlpTracesPath` and `otlpMetricsPath` constants in `endpoint.go`, used in both `tracing.go` and `metrics.go`.

File: pkg/telemetry/providers/otlp/metrics_test.go (hunk 4 above)
Comment (ChrisJBurns) [reply pkg/telemetry/providers/otlp/metrics_test.go:43]: Addressed in f5e03cd3. Added an error-path test case to `TestCreateMetricExporter` using an invalid CA cert path (`/nonexistent/ca.crt`), which exercises the `newTLSConfigFromCA` error branch. A cancelled-context approach doesn't work here because the OTLP metric HTTP SDK creates the exporter lazily (unlike the trace SDK), so construction succeeds even with a cancelled context.

File: pkg/telemetry/providers/otlp/endpoint.go [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+	idx := strings.Index(endpoint, "/")
+	if idx < 0 {
+		return endpoint, ""
+	}
+	return endpoint[:idx], strings.TrimRight(endpoint[idx:], "/")
```
[the remark this answers is not in the document]
Comment (ChrisJBurns) [reply pkg/telemetry/providers/otlp/endpoint.go:36]: Great analysis. Addressed in f5e03cd3 — `splitEndpointPath` now strips `http://` and `https://` prefixes defensively before looking for the first `/`. This makes it safe for both the operator path (where `NormalizeTelemetryConfig` already strips the scheme) and the CLI path (where it doesn't).

Added test cases covering the exact scenario described: `https://api.honeycomb.io` now correctly yields `hostPort="api.honeycomb.io"` and `basePath=""`.

**Review by reyortiz3 — APPROVED** [review]
